file/writer.go

```diff
@@ -581,13 +613,36 @@ func writeFile(content *Content, filename string, format Format) error {
 	}
 
 	if filename == "-" {
-		_, err = fmt.Print(string(c))
+		// TODO how should we write document files when output is stdout?
+		if _, err := fmt.Print(string(c)); err != nil {
+			return errors.Wrap(err, "writing file")
+		}
 	} else {
 		filename = utils.AddExtToFilename(filename, strings.ToLower(string(format)))
-		err = ioutil.WriteFile(filename, c, 0600)
-	}
-	if err != nil {
-		return errors.Wrap(err, "writing file")
+		prefix, _ := filepath.Split(filename)
+		if err := ioutil.WriteFile(filename, c, 0600); err != nil {
+			return errors.Wrap(err, "writing file")
+		}
+		for _, sp := range content.ServicePackages {
+			for _, d := range sp.Documents {
+				if err := os.MkdirAll(prefix+*sp.Name, 0700); err != nil {
+					return errors.Wrap(err, "creating document directory")
+				}
```

Comment:
Shouldn't this happen in the outer loop? You want to `mkdir` just once per service package, right? Probably with a `if len(sp.Documents) > 0`.

Reply:
Something of a leftover from an earlier understanding of the document objects (handled incorrectly, to boot).

Originally, our reading of the Document `Path` value was that it was actually a filesystem path, and that we'd need to create a nested structure under the SP/SV directories to handle the complete path. The original code didn't actually do this: handling those paths properly would require a `MkdirAll()` per document.

In practice, Konnect doesn't use `Path` like a filesystem path, and per confirmation from Henri P.:

> Paths are only used as display names

The rework in 575c8fa works off that understanding and creates the SP/SV directories only once, and only if they have an attached document.